blog/2022-02-08-turborepo-overhyped.md

```diff
@@ -0,0 +1,54 @@
+---
+slug: turborepo-overhyped
+title: Is Turborepo overhyped?
+authors: [sgranat]
+image: "/img/blog/turborepo.png"
+tags: [react, javascript, typescript, integration]
+---
+
+
+![Turborepo](/img/blog/turborepo.png)
+
+A two months ago I was loking for a solution how to speed up pipeline in our [monorepo](https://github.com/tolgee/tolgee-js). As an immediate answer I found basically two alternatives [Nx](https://nx.dev/) (from google) and [Turborepo](https://turborepo.org/) as a new cool project from well known opensource author ([Jared Palmer](https://github.com/jaredpalmer)). I was immediately drawn towards Turborepo as it was really hyped everywhere. Also it is quite minimalistic, simple to use and it looked like exact fit for our usecase. So I got into it.
+
+<!--truncate-->
+
+## What exactly is Turborepo and why we need it?
+
+Turborepo is a build system for monorepos (multiple projects in one repository). We use monorepo in Tolgee, because we have multiple SDKs for different frameworks as separate NPM packages and it is very convenient to test and release them together, so monorepo is an ideal fit.
+
+As with every monorepo we struggle to keep our building and testing pipeline fast. Turborepo is designed for this. Simply put, it keeps a cache of previous builds for each individual project (workspace) and if nothing changed in that project, it will just use the cache instead of rebuilding. It also automatically detects dependencies between the projects so it knows that if dependency of one project changed that project needs to be rebuild too.
+
+It can be also used for testing as you can only run tests in projects which are affected by changes and it treats standard output as part of a cache so even if some test don't run Turborepo will output cached logs, so you see all the test results every time.
```

Comment:
How does it detect which changes are related to the particular test?

Reply:
When there are changes in the project or in the project that is dependency